test/views/agent/YourIncomeSourceToSignUpViewSpec.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.agent
+
+import config.featureswitch.FeatureSwitch.ForeignProperty
+import forms.agent.BusinessIncomeSourceForm
+import forms.agent.BusinessIncomeSourceForm.incomeSourceKey
+import models.IncomeSourcesStatus
+import models.common.{OverseasProperty, SelfEmployed, UkProperty}
+import org.jsoup.Jsoup
+import org.jsoup.nodes.{Document, Element}
+import play.api.data.FormError
+import play.api.mvc.Call
+import utilities.UserMatchingSessionUtil.ClientDetails
+import utilities.ViewSpec
+import views.ViewSpecTrait
+import views.html.agent.YourIncomeSourceToSignUp
+
+class YourIncomeSourceToSignUpViewSpec extends ViewSpec {
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    disable(ForeignProperty)
+  }
+
+  object AgentIncomeSource {
+    val heading = "Your client’s income sources"
+    val paragraph1: String = "If your client is self-employed, you must add all of their sole trader businesses if they have more than one. " +
+      "If they have income from property you must add it, but this is limited to one UK property business."
+    val paragraph1Overseas: String = "Your client can have up to 50 sole trader businesses. " +
+      "However, they can have only one UK property business and one overseas property."
+    val paragraph2 = "Renting out a property includes using a letting agency."
+    val soleTrader = "Sole trader"
+    val soleTraderLinkText = "Add sole trader income source"
+    val soleTraderLink = appConfig.incomeTaxSelfEmploymentsFrontendClientInitialiseUrl
+    val ukProperty = "UK property"
+    val ukPropertyLinkText = "Add UK property income source"
+    val ukPropertyLink = "/report-quarterly/income-and-expenses/sign-up/client/business/property-commencement-date"
```

Comment:
```suggestion
    val ukPropertyLink = controllers.agent.business.routes.PropertyStartDateController.show().url
```
would ensure if the url ever needs changing in the future, this will automatically receive the same changes

same for foreign property link too